lib/createGltf.js

```diff
@@ -13,335 +16,902 @@ module.exports = createGltf;
+        accessor.byteOffset = byteLength;
+        byteLength += buffers[i].length;
+    }
+    gltf.bufferViews.push({
+        name : 'bufferView_' + bufferViewIndex,
+        buffer : 0,
+        byteLength : byteLength,
+        byteOffset : byteOffset,
+        byteStride : byteStride,
+        target : target
+    });
+}
+
+function addBuffers(gltf, bufferState) {
+    // Positions and normals share the same byte stride so they can share the same bufferView
+    var positionsAndNormalsAccessors = bufferState.positionAccessors.concat(bufferState.normalAccessors);
+    var positionsAndNormalsBuffers = bufferState.positionBuffers.concat(bufferState.normalBuffers);
+    addBufferView(gltf, positionsAndNormalsBuffers, positionsAndNormalsAccessors, 12, WebGLConstants.ARRAY_BUFFER);
+    addBufferView(gltf, bufferState.uvBuffers, bufferState.uvAccessors, 8, WebGLConstants.ARRAY_BUFFER);
+    addBufferView(gltf, bufferState.indexBuffers, bufferState.indexAccessors, undefined, WebGLConstants.ELEMENT_ARRAY_BUFFER);
+
+    var buffers = [];
+    buffers = buffers.concat(bufferState.positionBuffers, bufferState.normalBuffers, bufferState.uvBuffers, bufferState.indexBuffers);
+    var buffer = getBufferPadded(Buffer.concat(buffers));
+
+    gltf.buffers.push({
+        name : 'buffer',
+        byteLength : buffer.length,
+        extras : {
+            _obj2gltf : {
+                source : buffer
+            }
         }
-        return 'texture_' + getImageId(imagePath);
-    }
-
-    function createMaterial(material, hasNormals, options) {
-        var ambient = defaultValue(defaultValue(getTextureId(material.ambientTexture), material.ambientColor));
-        var diffuse = defaultValue(defaultValue(getTextureId(material.diffuseTexture), material.diffuseColor));
-        var emission = defaultValue(defaultValue(getTextureId(material.emissionTexture), material.emissionColor));
-        var specular = defaultValue(defaultValue(getTextureId(material.specularTexture), material.specularColor));
-        var alpha = defaultValue(defaultValue(material.alpha), 1.0);
-        var shininess = defaultValue(material.specularShininess, 0.0);
-        var hasSpecular = (shininess > 0.0) && (specular[0] > 0.0 || specular[1] > 0.0 || specular[2] > 0.0);
-
-        var transparent;
-        var transparency = 1.0;
-        if (typeof diffuse === 'string') {
-            transparency = alpha;
-            transparent = images[material.diffuseTexture].transparent || (transparency < 1.0);
-        } else {
-            diffuse[3] = alpha;
-            transparent = diffuse[3] < 1.0;
+    });
+}
+
+function getImage(images, imagePath, overrideImagePath, options) {
+    images = options.overridingImages.concat(images);
+    imagePath = defaultValue(overrideImagePath, imagePath);
+    var imagesLength = images.length;
+    for (var i = 0; i < imagesLength; ++i) {
+        var image = images[i];
+        if (image.path === imagePath) {
```

Comment:
I started to make the helper function but realized `getTexture` would need slightly different behavior since it wants the index instead of the object. Then I decided it might not be worth it.